src/blockchain/src/notifier.rs

```diff
@@ -7,15 +7,16 @@ use crate::blockchain::Blockchain;
 #[derive(Hash, Eq, PartialEq, Clone)]
 pub struct BlockchainError;
 
-pub type BlockchainResult = std::result::Result<(), BlockchainError>;
-pub type OnTransactionSettled = Box<dyn FnOnce(Transaction)>;
-pub type OnBlockEvent = Box<dyn FnMut(Block)>;
-
 // But we require certain bounds to get things done...
 impl Blockchain {
     // should we borrow or own this transaction?
-    pub fn submit_transaction(&mut self, trans: Transaction, on_done: OnTransactionSettled) {
-        self.trans_observers.insert(trans, on_done);
+    pub fn submit_transaction<CB: 'static + FnOnce(Transaction) + FnOnce(Transaction)>(
```

Comment:
I dont' think `FnOnce(Transaction)` twice is needed. I will try to remove the extra

Reply:
Name is a little confusing

```suggestion
    pub fn submit_transaction<callback: 'static + FnOnce(Transaction) + FnOnce(Transaction)>(
```